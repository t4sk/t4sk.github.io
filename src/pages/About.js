import React from "react"

export function About(prop) {
  return <div>About</div>
}

export default About
