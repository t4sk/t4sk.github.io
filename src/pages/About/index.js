import React from "react"
import { Image, Icon } from "semantic-ui-react"
import PageLayout from "../../components/PageLayout"
import MainLayout from "../../components/MainLayout"
import SEO from "../../components/SEO"
import styles from "./index.module.css"
import profile from "./profile.jpeg"

function About(props) {
  return (
    <PageLayout>
      <SEO title="About | Smart Contract Programmer" />
      <MainLayout>
        <div className={styles.Component}>
          <Image
            src={profile}
            circular
            fluid
            style={{ width: 150, height: 150 }}
            alt="profile"
          />
          <h1 className={styles.header}>Tasuku Nakamura</h1>
          <div className={styles.links}>
            <a
              href="https://www.youtube.com/channel/UCJWh7F3AFyQ_x01VKzr9eyA"
              target="__blank"
              rel="noopener noreferrer"
            >
              <Icon name="youtube" color="red" size="huge" />
            </a>
            <a
              href="https://twitter.com/ProgrammerSmart"
              target="__blank"
              rel="noopener noreferrer"
            >
              <Icon name="twitter" color="blue" size="huge" />
            </a>
            <a
              href="https://github.com/t4sk"
              target="__blank"
              rel="noopener noreferrer"
            >
              <Icon name="github" color="black" size="huge" />
            </a>
          </div>
          <p className={styles.intro}>
            Hi, I am a software engineer living in Japan, specializing in smart
            contracts. <br />
            You can contact me at{" "}
            <span className={styles.email}>
              contact@smartcontractprogrammer.com
            </span>
          </p>
        </div>
      </MainLayout>
    </PageLayout>
  )
}

About.propTypes = {}

export default About
