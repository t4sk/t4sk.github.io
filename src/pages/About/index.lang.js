const LANG = {
  en: {
    About: "About",
    Contact: "Contact",
  },
  jp: {
    About: "概要",
    Contact: "お問い合わせ",
  },
}

export default LANG
