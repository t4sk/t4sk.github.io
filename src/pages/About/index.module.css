.Component {
  display: flex;
  flex-direction: column;
  align-items: center;
}
.header {
  font-size: 2.5rem;
}
.links {
  display: flex;
  flex-direction: row;
}
.intro {
  color: #4e4e4e;
  font-size: 1.3rem;
  margin-top: 50px;
}
.email {
  font-weight: 700;
}
