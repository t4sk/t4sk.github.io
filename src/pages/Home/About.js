import React from "react"
import { Image, Icon, Divider } from "semantic-ui-react"
import styles from "./About.module.css"
import profile from "./profile.jpeg"

function About(props) {
  // TODO consulting and contact form

  return (
    <div className={styles.component}>
      <div className={styles.profile}>
        <Image
          src={profile}
          circular
          fluid
          style={{ width: 150, height: 150 }}
          alt="profile"
        />
        <h1 className={styles.header}>Tasuku Nakamura</h1>
      </div>
      <div className={styles.intro}>
        <p>
          Hi, I am a software engineer who makes videos about smart contracts.
        </p>
        <p>
          Creator of{" "}
          <a
            href="https://solidity-by-example.org/"
            target="__blank"
            rel="noopener noreferrer"
          >
            Solidity by Example
          </a>
        </p>
      </div>

      <Divider />

      <h3>Follow me</h3>
      <div>
        <a
          href="https://www.youtube.com/channel/UCJWh7F3AFyQ_x01VKzr9eyA/featured"
          target="__blank"
          rel="noopener noreferrer"
        >
          <div className={styles.followMeLink}>
            <Icon name="youtube" color="red" size="large" />
            YouTube
          </div>
        </a>
        <a
          href="https://twitter.com/ProgrammerSmart"
          target="__blank"
          rel="noopener noreferrer"
        >
          <div className={styles.followMeLink}>
            <Icon name="twitter" color="blue" size="large" />
            Twitter
          </div>
        </a>
        <a
          href="https://github.com/t4sk"
          target="__blank"
          rel="noopener noreferrer"
        >
          <div className={styles.followMeLink}>
            <Icon name="github" color="black" size="large" />
            GitHub
          </div>
        </a>
      </div>

      <Divider />
      <p>
        You can contact me at{" "}
        <span className={styles.email}>
          contact@smartcontractprogrammer.com
        </span>
      </p>
    </div>
  )
}

About.propTypes = {}

export default About
