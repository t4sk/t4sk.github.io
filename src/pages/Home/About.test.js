import React from "react"
import { shallow } from "enzyme"
import About from "./About"

const props = {}

test("it renders", () => {
  const component = shallow(<About {...props} />)

  expect(component).toMatchSnapshot()
})
