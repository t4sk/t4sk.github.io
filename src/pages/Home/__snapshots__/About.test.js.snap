// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<div
  className="component"
>
  <div
    className="profile"
  >
    <Image
      alt="profile"
      as="img"
      circular={true}
      fluid={true}
      src="profile.jpeg"
      style={
        Object {
          "height": 150,
          "width": 150,
        }
      }
      ui={true}
    />
    <h1
      className="header"
    >
      Tasuku Nakamura
    </h1>
  </div>
  <div
    className="intro"
  >
    <p>
      Hi, I am a software engineer who makes videos about smart contracts.
    </p>
    <p>
      Creator of
       
      <a
        href="https://solidity-by-example.org/"
        rel="noopener noreferrer"
        target="__blank"
      >
        Solidity by Example
      </a>
    </p>
  </div>
  <Divider />
  <h3>
    Follow me
  </h3>
  <div>
    <a
      href="https://www.youtube.com/channel/UCJWh7F3AFyQ_x01VKzr9eyA/featured"
      rel="noopener noreferrer"
      target="__blank"
    >
      <div
        className="followMeLink"
      >
        <Icon
          as="i"
          color="red"
          name="youtube"
          size="large"
        />
        YouTube
      </div>
    </a>
    <a
      href="https://twitter.com/ProgrammerSmart"
      rel="noopener noreferrer"
      target="__blank"
    >
      <div
        className="followMeLink"
      >
        <Icon
          as="i"
          color="blue"
          name="twitter"
          size="large"
        />
        Twitter
      </div>
    </a>
    <a
      href="https://github.com/t4sk"
      rel="noopener noreferrer"
      target="__blank"
    >
      <div
        className="followMeLink"
      >
        <Icon
          as="i"
          color="black"
          name="github"
          size="large"
        />
        GitHub
      </div>
    </a>
  </div>
  <Divider />
  <p>
    You can contact me at
     
    <span
      className="email"
    >
      contact@smartcontractprogrammer.com
    </span>
  </p>
</div>
`;
